lib/3scale/backend/listener.rb

```diff
@@ -582,6 +577,20 @@ def check_post_content_type!
         raise ContentTypeInvalid, ctype if invalid_post_content_type?(ctype)
       end
 
+      def check_transactions_validity(transactions)
+        if blank?(transactions)
+          raise TransactionsIsBlank
+        end
+
+        if !transactions.is_a?(Hash)
+          raise TransactionsFormatInvalid
+        end
+
+        if transactions.any? { |_id, data| data.nil? }
+          raise TransactionsHasNilTransaction
```

Comment:
can this actually happen? all the cases I imagine where a hash has a key the "data"/value part of it should not be nil but at most an empty string.

Reply:
This can happen.

For example, I reproduced the problem by performing the following request (see the last transactions specified without values):

```
[<username>@localhost apisonator]$ curl -X POST -v '127.0.0.1:3000/transactions.xml' -d 'provider_key=provider_keytest&transactions[0][app_id]=5207&transactions[0][usage][hits]=1&transactions[0][timestamp]=2018-04-17 10:13:11 UTC&transactions[1]'
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 3000 (#0)
> POST /transactions.xml HTTP/1.1
> Host: 127.0.0.1:3000
> User-Agent: curl/7.55.1
> Accept: */*
> Content-Length: 156
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 156 out of 156 bytes
< HTTP/1.1 400 Bad Request
< Content-Type: application/vnd.3scale-v2.0+xml
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: ETag, Link, 3scale-rejection-reason
< X-Content-Type-Options: nosniff
< Transfer-Encoding: chunked
< 
* Connection #0 to host 127.0.0.1 left intact
<?xml version="1.0" encoding="UTF-8"?><error code="transactions_has_nil_transaction">transactions has a nil transaction</error>[<username>@localhost apisonator]$ 
```

The test 'returns error if any of the reported transactions are nil' in test/integration/report_test.rb also reproduces the nil error message.

Reply:
ok :+1: